overlord/snapstate/snapstate.go

```diff
@@ -471,6 +472,39 @@ func CheckChangeConflict(st *state.State, snapName string, checkConflictPredicat
 	return nil
 }
 
+// GuardCoreSetupProfilesPhase2 decides for a setup-profiles task that
+// is a phase 2 security setup for core whether to proceed, wait
+// (via Retry) or there is nothing to do. Helper for ifacestate.
+func GuardCoreSetupProfilesPhase2(task *state.Task, snapsup *SnapSetup, snapInfo *snap.Info) (proceed bool, err error) {
+	// phase 2 setup-profiles
+	if snapInfo.Type != snap.TypeOS {
```

Comment:
This is not happening in practice, right? I.e. usually we only call this helper if core-phase-2 is true. Still good to have this check.

Reply:
we always had this check, I don't remember the details but InstallPath always adds the 2nd setup-profile